### PR TITLE
Use `*_t` and `*_v` trait helpers from C++17 STL

### PR DESCRIPTION
**Pull request by louwers**
Progress towards #335

@wipfli Do you have any other C++17 cleanup ideas in mind?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -40,3 +40,4 @@ node_modules
 /lib
 
 /bazel-*
+CMakeLists.txt.user
```
Comment (ntadej) [inline .gitignore:43]: This should usually be a separate PR 😊

File: include/mbgl/actor/actor_ref.hpp [hunk 2]
```diff
@@ -38,11 +38,7 @@ class ActorRef {
     template <typename Fn, class... Args>
     auto ask(Fn fn, Args&&... args) const {
         // Result type is deduced from the function's return type
-#if __cplusplus >= 201703L
```
Comment (ntadej) [inline include/mbgl/actor/actor_ref.hpp:41]: This will certainly break some platforms (unless we fixed them).

**Comment by ntadej** [comment]
I'm not sure we actually moved all the clients to C++17.

We should also remove all the polyfill libraries that were made/added by Mapbox to get C++17 features for older compilers.

File: include/mbgl/actor/actor_ref.hpp (hunk 2 above)
Comment (ntadej) [reply include/mbgl/actor/actor_ref.hpp:41]: As expected, the macOS build fails.

**Comment by louwers** [comment]
> As expected, the macOS build fails.

Hmm, I will check out why it doesn't use C++17. It is set in the root `CMakeLists.txt`, but it probably needs to be set in the sub project as well.

> We should also remove all the polyfill libraries that were made/added by Mapbox to get C++17 features for older compilers.

@ntadej Thanks, I will have a look! Do you know anything about those polyfill libraries? Are they in `vendor`?

**Comment by louwers** [comment]
Neat, macos passes now.

**Comment by ntadej** [comment]
Great! It may be useful to check for additional warnings though. Not sure if we allow them or not.

**Comment by louwers** [comment]
I had to add CMAKE_CXX_STANDARD and CMAKE_CXX_STANDARD_REQUIRED to platform/android/MapboxGLAndroidSDK/src/cpp/CMakeLists.txt

This file is actually used as a root CMakeLists.txt and uses `add_subdirectory`, which discards the variable scope upon re-entry. It's a bit weird that the root CMakeLists.txt is used both as a child and in other places as a parent.

Another problem is that the CMakeLists.txt under MapboxGLAndroidSDK does not call project() which it should as a root.

**Comment by ntadej** [comment]
Yes, these extra CMake files should go away or be simplified with the reorganisation.

**Comment by ntadej** [comment]
Hi @louwers, what if we merge this now and remove polyfills in a separate one? No need to do everything at once 😊

**Comment by louwers** [comment]
@ntadej Sure!

**Review by ntadej — APPROVED** [review]

**Comment by wipfli** [comment]
> @wipfli Do you have any other C++17 cleanup ideas in mind?

I don't know
